iac-extensions/jvm-framework-config/src/main/java/org/sonar/iac/jvmframeworkconfig/checks/micronaut/HardcodedSecretsCheck.java

```diff
@@ -0,0 +1,210 @@
+    "micronaut.security.token.jwt.generator.refresh-token.secret",
+    "micronaut.security.token.jwt.claims-validators.openid-idtoken",
+    "jasync.client.password",
+    "vertx.mysql.client.password",
+    "vertx.pg.client.password",
+    "tracing.zipkin.http.proxy-password",
+    "tracing.jaeger.sender.auth-password",
+    "otel.exporter.zipkin.proxy-password");
+  private static final List<String> SENSITIVE_KEYS_REGEX = List.of(
+    "micronaut.http.services.[^.]++.proxy-password",
+    "micronaut.http.services.[^.]++.ssl.key.password",
+    "micronaut.http.services.[^.]++.ssl.key-store.password",
+    "micronaut.http.services.[^.]++.ssl.trust-store.password",
+    "micronaut.chatbots.telegram.bots.[^.]++.token",
+    "flyway.datasources.[^.]++.password",
+    "nats.[^.]++.password",
+    "nats.[^.]++.token",
+    "nats.[^.]++.tls.trust-store-password",
+    "r2dbc.datasources.[^.]++.password",
+    "rabbitmq.servers.[^.]++.password",
+    "redis.servers.[^.]++.password",
+    "micronaut.security.token.jwt.encryptions.secret.[^.]++.secret",
+    "micronaut.security.token.jwt.signatures.secret.[^.]++.secret",
+    "micronaut.security.ldap.[^.]++.context.manager-password",
+    "micronaut.security.oauth2.clients.[^.]++.client-secret",
+    "datasources.[^.]++.password");
+  private static final Pattern PATTERN_PASSWORD_IN_CONNECTION_STRING = Pattern.compile("(?:;|^)AccountKey=(?<password>[a-zA-Z0-9+/=]{60,})(?:;|$)");
+  private static final Pattern PATTERN_PASSWORD_IN_URL = Pattern.compile("(?<protocol>[a-zA-Z]++)://(?<username>[^:@]++):(?<password>[^@]++)@.++");
+  private static final Map<String, Pattern> SENSITIVE_KEYS_WITH_PATTERN_VALUE = Map.of(
+    "azure.credential.storage-shared-key.connection-string", PATTERN_PASSWORD_IN_CONNECTION_STRING,
+    "micronaut.jms.activemq.artemis.connection-string", PATTERN_PASSWORD_IN_URL,
+    "micronaut.jms.activemq.classic.connection-string", PATTERN_PASSWORD_IN_URL,
+    "mqtt.client.server-uri", PATTERN_PASSWORD_IN_URL,
+    "rabbitmq.uri", PATTERN_PASSWORD_IN_URL);
+  // Extra use case: combine both wildcard in the key and URL with password in value
+  private static final Pattern PATTERN_SENSITIVE_RABBITMQ_PROPERTY = Pattern.compile("rabbitmq.servers.[^.]++.uri");
+
+  private static final Pattern VARIABLE = Pattern.compile("\\$\\{[^}]+}");
+  // Compiling patterns together is ~35% faster than checking patterns individually
+  private static final Predicate<String> PREDICATE_SENSITIVE_KEY_FULL_REGEX = Pattern.compile(String.join("|", SENSITIVE_KEYS_REGEX)).asMatchPredicate();
+
+  @Override
+  protected Set<String> sensitiveKeys() {
+    return SENSITIVE_KEYS;
+  }
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(Tuple.class, this::checkTuplePatternInKey);
+    init.register(Tuple.class, HardcodedSecretsCheck::checkTuplePatternInValue);
+    super.initialize(init);
+  }
+
+  private void checkTuplePatternInKey(CheckContext ctx, Tuple tuple) {
```

Comment:
This could also be an overriding method of `checkTuple` from the `AbstractSensitiveKeyCheck` class

Reply:
Do you mean to have the equivalent of `sensitiveKeys`/`checkTuple` but for patterns in `AbstractSensitiveKeyCheck`?

I have a mixed feeling about it.
I did this initially by adding an abstract method `sensitivePatterns();`, but it forced me to add an implementation returning an empty list in the 6 other rules.
Because of this, I preferred to move this logic directly in the rule as currently it is the only one using it.
I was thinking that If later we need this logic in another (new) rule, we could move it to the `AbsractSensitiveKeyCheck` layer. But it implies that we are aware of its existence in `HardcodedSecretsCheck` and do not miss it.

WDYT?

Reply:
No, just overriding the checkTuple method in this check, and the content of the method will be everything in checkTuplePatternInKey

Reply:
Got it 👍 it indeed simplify the logic and embed well with your other comment, nice catch!